src/simulation/decision_model_simulation_results.jl

```diff
@@ -305,20 +254,25 @@ function _process_timestamps(
 end
 
 function _read_results(
-    T::Type{Matrix{Float64}},
+    ::Type{Matrix{Float64}},
     res::SimulationProblemResults{DecisionModelSimulationResults},
     result_keys,
     timestamps::Vector{Dates.DateTime},
-    store::Nothing,
+    store::Union{Nothing, <:SimulationStore},
```

Comment:
This weird interface relates to #1018 and could be fixed in this PR. @daniel-thom we discussed not doing this anymore.

Reply:
I'm not sure exactly what the proposal is here. For now I'm just mimicking the other `_read_results` method and passing the store straight through

Reply:
This is an interface issue @daniel-thom discussed where we pass a store to read from or nothing when nothing signifies that it is an in-memory store which makes the code confusing since it isn't obvious.

Reply:
So you'd like me to resolve the `store` to the in-memory store or not somewhere upstream so there's always a valid store getting passed around?

Reply:
I’m not sure that I would change this, at least not here. `nothing` also means that the user didn’t pass a store.

Reply:
what does not passing a store to get result mean? I don't think that is right

Reply:
We could easily be talking around each other. There are many nuances here. The user can do either of these:
```
read_variable(results, “ActivePowerVariable__ThermalStandard”)
open_store(HdfSimulationStore, simulation_store_path, "r") do store
    read_variable(results, “ActivePowerVariable__ThermalStandard”; store=store)
end
```
In the first case we have to get the store, which could in memory or h5. If it’s h5, we have to open it. It’s in-memory, we can just use it. There is one slightly awkward case one line 288.

Reply:
no one does this 

```
open_store(HdfSimulationStore, simulation_store_path, "r") do store
    read_variable(results, “ActivePowerVariable__ThermalStandard”; store=store)
end
```

Assuming this patter will happen make the code way less understandable for developers. 

The most common way to access results is calling  `read_variables` or `read_realized_variables`

Reply:
In the absence of consensus, are we okay keeping `store::Union{Nothing, <:SimulationStore}` in this non-user-facing method for the moment?

Reply:
I’ll propose a patch in a separate PR.